### PR TITLE
Template cohesion

### PR DESCRIPTION
**Pull request by killerswin2**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    oh boi. This Changes alot. 
    1) moves all templated related functionality to a new addon.
    2) a lot documentation, just about every function now has documentation.
    3) templates no longer have unit type definitions, those were moved to one function
    4) loading templates now has an event, allowing 3rd party mods to get the hashmap before loadout creation.
    5) templates now allow have facewear instead of just one vanilla templates.
    6) verification of templates now use the new compatableItems and magazines 
    7) fixed a the vanilla templates trying to add a optic to a rail slot instead of an optic slot.
    8) verification now checks what type of item you are trying you add if it fails.
    9) fixed casing in some cup templates
    10) fixed bad optic names in baf cup templates
    11) fixed adding a rpg mag in a m72
    12) private functions now are prefixed if they weren't
    13) template folder that holds the templates now is named modsets, like addons
    
    Why all these changes, started for testing documentation they decided to make the templates a little cleaner and more dry.
    This is the first step in allowing templates to be json as now we don't need to have loadout generation code in templates.
    There is probably something I missed in explaining but that should be it. Test with every modset. 

### Please specify which Issue this PR Resolves.
N/A

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Review by Lazejun — CHANGES_REQUESTED** [review]
Don't add Facewear unless you do it properly.

Dont use the horrible existing example as a base.

File: A3A/addons/Templates/Templates/Modsets/IFA/IFA_AI_POL_Arid.sqf [hunk 1]
```diff
@@ -0,0 +1,236 @@
+//////////////////////////
+//   Side Information   //
+//////////////////////////
+
+["name", ""] call _fnc_saveToTemplate;
+["spawnMarkerName", ""] call _fnc_saveToTemplate;
```
Comment (Lazejun) [inline A3A/addons/Templates/Templates/Modsets/IFA/IFA_AI_POL_Arid.sqf:6]: Just delete these dead files

File: A3A/addons/Templates/functions/Templates/functionTemplates.sqf [hunk 2]
```diff
@@ -0,0 +1,689 @@
...
+        ["Grenadier", _grenadierTemplate],
+        ["LAT", _latTemplate],
+        ["AT", _atTemplate],
+        ["AA", _aaTemplate],
+        ["MachineGunner", _machineGunnerTemplate],
+        ["Marksman", _marksmanTemplate],
+        ["Sniper", _sniperTemplate]
+    ];
+ 
+    _templatesToGenerate pushBack [_prefix, _unitTypes, "sfLoadoutData"];
```
Comment (HakonRydland) [inline A3A/addons/Templates/functions/Templates/functionTemplates.sqf:511]: adding the templates to list of templates should be accessible to the template so they can change stuff, the base unit template can remain where it is as then you could just duplicate and overwrite it, i dont believe this is to much boiler plate to leave in

**Review by HakonRydland — CHANGES_REQUESTED** [review]
wont speak to much on this, but i do see one concern, at a glance the rest looks fine but as its mostly template data im leaving that for laze, rest of changes is mostly just moving stuff around, seeing as its a rather big and isolated system im fine with this becoming its own addon

**Comment by killerswin2** [comment]
this pr has been broken up into new prs

**Pull request closed without merging**